edx2bigquery/rephrase_tracking_logs.py

```diff
@@ -278,6 +278,27 @@ def check_empty(data, *keys):
 
     data.pop('event_js')	# leftover from mongo import script
 
+    #-----------------------------------------
+    # check for null values in speed_change_video
+    # Error encountered parsing LAC data from Oct. 2013
+    # Requires that we also be able to convert the value to a float
+
+    def string_is_float(s):
+        try:
+            float(s)
+            return True
+        except ValueError:
+            return False
+
+    if data['event_type']=='speed_change_video':
+        if 'event_struct' in data and 'new_speed' in data['event_struct']:
+            # First check if string is float
+            if string_is_float(data['event_struct']['new_speed']):
+                # Second check if value is null
+                if float(data['event_struct']['new_speed']) != float(data['event_struct']['new_speed']):
```

Comment:
When float(x) = nan

I usually do this with "pd.isnull" or "math.isnan", but did not want to
load another package. If you think it would be cleaner with math, I can use
a built in function.

http://stackoverflow.com/questions/944700/how-to-check-for-nan-in-python

My understanding is that float(x) != float(x) is a typical way to check for
NaN.

Best,
Daniel

On Thu, Oct 22, 2015 at 1:52 PM, ichuang <notifications@github.com> wrote:

> In edx2bigquery/rephrase_tracking_logs.py
> <https://github.com/mitodl/edx2bigquery/pull/36#discussion_r42781563>:
>
> > +    # Error encountered parsing LAC data from Oct. 2013
> > +    # Requires that we also be able to convert the value to a float
> > +
> > +    def string_is_float(s):
> > +        try:
> > +            float(s)
> > +            return True
> > +        except ValueError:
> > +            return False
> > +
> > +    if data['event_type']=='speed_change_video':
> > +        if 'event_struct' in data and 'new_speed' in data['event_struct']:
> > +            # First check if string is float
> > +            if string_is_float(data['event_struct']['new_speed']):
> > +                # Second check if value is null
> > +                if float(data['event_struct']['new_speed']) != float(data['event_struct']['new_speed']):
>
> For what x does float(x) != float(x) ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mitodl/edx2bigquery/pull/36/files#r42781563>.
>